### PR TITLE
Update Team and make Teams a 1-element list

### LINKED ISSUE #7355: Clean up Teams from the system
As discussed with Seangchan yesterday (and initially fixed in #7353), we're no longer supporting a workflow assigned to more than one team name. 

In addition to that, we'll also remove the support of multiple team names defined in the agent configuration. It just causes confusion about `Team` x `Teams` and nobody has used this feature during the last couple of years.

### LINKED ISSUE #8074: Team = "" but Teams = ["production"]
looking at 

https://cmsweb.cern.ch/reqmgr2/fetch?rid=vlimant_task_MUO-PhaseIITDRSpring17DR-00001__v1_T_170817_194940_3043

and related to recent issues,

  "Team": "", 

while 

  "Teams": [
    "production"
  ], 

 shouldn't "Team" = "production"

not an issue, it seems since the WQE end up with production as TeamName, but does not look too consistent to me.

### PR DESCRIPTION
**Pull request by amaltaro**
Partially fixes #7355 
Fixes #8074 

Removing only part of the Teams logic for now, otherwise we'd need to make several codes handling it both ways (Team and Teams). So I'd rather make sure Team has the proper value and then in the next month we proceed with the final cleanup.

Real change is on updaterequest.js, such that we:
* properly set `Team` to the string provided during assignment
* set `Teams` to the same value as Team, but list type

in addition to that:
* assumes the agent can only be connected to a single team name
* removed `InputDatasetTypes`, I guess it has never been used

### CODE REVIEW

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
@ericvaandering in case you find some spare cycles, could you please have a look at jenkins? It seems it's hiting an "internal" failure and then it gets automatically restarted. Giving up after some retries without any error/message.

**Comment by ericvaandering** [comment]
I was actually looking at this an hour or so ago. Seems to be nodes getting full of old docker junk, so I cleaned the worst 10 or so offenders.

**Comment by ericvaandering** [comment]
test this please

**Comment by amaltaro** [comment]
awesome, thanks Eric!

**Comment by ticoann** [comment]
We have to double check when team is assigned through reqmr2 GUI, this still works.

**Comment by amaltaro** [comment]
Good point! I tested all possible Team values via script and the validation works fine (including the workload couch document). I'll do the same via web UI and update it here.

**Review by ticoann — APPROVED** [review]

File: src/couchapps/ReqMgr/updates/updaterequest.js [hunk 1]
```diff
@@ -17,15 +17,12 @@ function(doc, req) {
...
+
     function updateTeams(team) {
-        if (!doc.Teams) {
-            doc.Teams = new Array();
-            doc.Teams.push(team);
-        } else {
-            doc.Teams.push(team);
-        }
+        doc.Teams = new Array();
+        doc.Teams.push(team);
```
[the remark this answers is not in the document]
Comment (amaltaro) [reply src/couchapps/ReqMgr/updates/updaterequest.js]: We could, but I rather trust that the StdBase validation will be enough and not have duplicate code.
If you insist, I can add it back :-)

**Comment by amaltaro** [comment]
Seangchan, I just had a look at assignment via web UI and it's even better, since we have a drop-down menu it only allow us to select an existent team or the default selection (which is nothing=empty string). Also correctly fails assignment if no team is select

From my side, it's good to go.

File: src/python/WMCore/ReqMgr/Service/Request.py [hunk 2]
```diff
@@ -417,6 +415,11 @@ def _handleAssignmentStateTransition(self, workload, request_args, dn):
 
         # by default, it contains all unmerged LFNs (used by sites to protect the unmerged area)
         request_args['OutputModulesLFNBases'] = workload.listAllOutputModulesLFNBases()
+
+        # FIXME: remove it on HG1710, when this #7355 is complete fixed
```
Comment (amaltaro) [inline src/python/WMCore/ReqMgr/Service/Request.py]: @ticoann Seangchan, as we discussed, I added this additional Team validation that we can remove in the next month or so

**Comment by amaltaro** [comment]
test this please

**Review by ticoann — APPROVED** [review]

**Pull request merged**